findme-location-riddles/src/LocationRiddlesService.py

```diff
@@ -3,32 +3,50 @@
 import boto3
 import os
 
+from typing import Union
+from decimal import Decimal
+from pydantic import ValidationError
 from aws_lambda_powertools.event_handler.exceptions import (
     BadRequestError,
     NotFoundError,
 )
 from src.helpers.CalculateScore import calculate_score_and_distance
+from src.entities.LocationRiddle import LocationRiddle, LocationRiddleDTO, SolvedLocationRiddleDTO
+from src.entities.Rating import Rating
+from src.entities.Comment import Comment
+from src.entities.Guess import Guess
 
 
 class LocationRiddlesService:
     def __init__(self, location_riddle_repository, image_bucket_repository):
         self.image_bucket_repository = image_bucket_repository
         self.location_riddle_repository = location_riddle_repository
 
-    def post_location_riddle(self, image_base64, location, user_id):
-        location_riddle_id = str(uuid.uuid4())
-        image_path = f"location-riddles/{location_riddle_id}.png"
+    def post_location_riddle(self, image_base64: str, location: list, user_id: str) -> dict:
+        location_riddle_data = {
+            "location_riddle_id": str(uuid.uuid4()),
+            "user_id": user_id,
+            "location": [Decimal(str(coord)) for coord in location]
+        }
 
         try:
-            self.location_riddle_repository.write_location_riddle_to_db(
-                user_id, location_riddle_id, location
+            location_riddle = LocationRiddle(**location_riddle_data)
+        except ValidationError as e:
+            print(f"unable to update location_riddle with provided parameters. {e}")
+            raise BadRequestError(
+                f"unable to update location_riddle with provided parameters. {e}"
             )
+
+        try:
+            self.location_riddle_repository.write_location_riddle_to_db(location_riddle)
         except Exception as e:
-            raise BadRequestError(f"{e}")
+            print(e)
+            raise BadRequestError(e)
 
+        image_path = f"location-riddles/{location_riddle.location_riddle_id}.png"
         return self.image_bucket_repository.post_image_to_s3(image_base64, image_path)
 
-    def get_location_riddle(self, location_riddle_id):
+    def get_location_riddle(self, location_riddle_id: str, user_id: str) -> Union[LocationRiddleDTO, SolvedLocationRiddleDTO]:
```

Comment:
minor & just FYI: I think we could use the new syntax for typing Unions
https://www.blog.pythonlibrary.org/2021/09/11/python-3-10-simplifies-unions-in-type-annotations/
```suggestion
    def get_location_riddle(self, location_riddle_id: str, user_id: str) -> LocationRiddleDTO | SolvedLocationRiddleDTO:
```

⚠️ should be tested not just committed